ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -317,19 +327,21 @@ def timeout(self, func, seconds, message):
 
     def generate_vlan_servers(self):
         vlan_host_map = defaultdict(dict)
-        vlan_ip_range = self.test_params['vlan_ip_range']
-
-        _, mask = vlan_ip_range.split('/')
-        n_hosts = min(2**(32 - int(mask)) - 3, self.max_nr_vl_pkts)
+        self.nr_vl_pkts = 0
+        for vlan, prefix in self.vlan_ip_range.items():
+            if not self.ports_per_vlan[vlan]:
+                continue
+            _, mask = prefix.split('/')
+            n_hosts = min(2**(32 - int(mask)) - 3, self.max_nr_vl_pkts)
 
-        for counter, i in enumerate(xrange(2, n_hosts + 2)):
-            mac = self.VLAN_BASE_MAC_PATTERN.format(counter)
-            port = self.vlan_ports[i % len(self.vlan_ports)]
-            addr = self.host_ip(vlan_ip_range, i)
+            for counter, i in enumerate(xrange(2, n_hosts + 2)):
+                mac = self.VLAN_BASE_MAC_PATTERN.format(counter)
+                port = self.ports_per_vlan[vlan][i % len(self.ports_per_vlan[vlan])]
+                addr = self.host_ip(prefix, i)
 
-            vlan_host_map[port][addr] = mac
+                vlan_host_map[port][addr] = mac     # For simplicity, assuming there is no overlapped ports between vlans
```

Comment:
>assuming [](http://example.com/codeflow?start=70&length=8)

Why assume this?

Reply:
I checked this part again, it seems that it does not necessarily require the assumption. Removed the comment.